contracts/factory/src/storage.rs

```diff
@@ -68,6 +76,16 @@ pub fn save_lp_vec(env: &Env, lp_info: Vec<Address>) {
     env.storage().instance().set(&DataKey::LpVec, &lp_info);
 }
 
+pub fn save_lp_vec_with_tuple_as_key(env: &Env, tuple_pair: &TokenInitInfo, lp_address: &Address) {
```

Comment:
Reference is cheap but it's still an overkill to pass the whole structure, it's enough to pass `(Address, Address)` or something similar - then you won't have to import the whole structure.